### DISCUSSION TITLE
The number of types atom and the order of atom in DP repeated training

### DISCUSSION THREAD

**Discussion by jun199615** [deepmodeling/deepmd-kit#3990] [Q&A]
Hello, everyone. 
1. If initially a potential function with 3 elements is generated using DeePMD-kit, then a dataset with four elements is re-trained based on this potential function, can this be achieved, and are there any keywords that need to be changed in the input.json?
When I use this "dp train -f frozen_model.pb input.json", i face a error "ValueError: could not broadcast input array from shape (4,128) into shape (30,128)". It seems to remind me that I can't add new elements for retraining base on my frozen_model.pb only contain 3 atom types.
2. When there are multiple AIMD datasets（from VASP）, do the types and  orders of atoms in these datasets have an effect on the training results? (Although my training test jobs run successfully)

**Comment by njzjz** [comment]
> 1. If initially a potential function with 3 elements is generated using DeePMD-kit, then a dataset with four elements is re-trained based on this potential function, can this be achieved, and are there any keywords that need to be changed in the input.json?
>        When I use this "dp train -f frozen_model.pb input.json", i face a error "ValueError: could not broadcast input array from shape (4,128) into shape (30,128)". It seems to remind me that I can't add new elements for retraining base on my frozen_model.pb only contain 3 atom types.

In theory, it can, though it is tough to do so. Contribution is welcome.

>     2. When there are multiple AIMD datasets（from VASP）, do the types and  orders of atoms in these datasets have an effect on the training results? (Although my training test jobs run successfully)

The orders of atoms don't affect the model output. I am not sure what you mean "types".
